### PR TITLE
Fix compile warnings and add CI windows tests

### LINKED ISSUE #165: Compilation warning with clang ++
`xxx/build/geodiff/1.0.6/geodiff/src/changeset.h:65:5: warning: non-void function does not return a value in all control paths [-Wreturn-type]
    }
    ^
`

### PR DESCRIPTION
**Pull request by PeterPetrik**
- added new CMAKE option "PEDANTIC=TRUE(default)"
- fixed some compilation warnings coming from PEDANTIC
- added cppcheck to CI, fix cppcheck warnings
- add windows build&test to the mix 
- it looked like we used sqlite3 on windows without RTREE, so completely broken :(. Fortunately we replaced it with QGIS sqlite3 from OSGeo4w for plugin, so in QGIS Plugin it worked.

fix #165

### CODE REVIEW

File: geodiff/src/changesetvarint.h [hunk 1]
```diff
@@ -217,7 +217,7 @@ static u8 sqlite3GetVarint( const unsigned char *p, u64 *v )
 ** single-byte case.  All code should use the MACRO version as
 ** this function assumes the single-byte case has already been handled.
 */
-static u8 sqlite3GetVarint32( const unsigned char *p, u32 *v )
+static inline u8 sqlite3GetVarint32( const unsigned char *p, u32 *v )
```
Comment (wonder-sk) [inline geodiff/src/changesetvarint.h:220]: both static and inline?

File: geodiff/src/geodiff-cli.cpp [hunk 2]
```diff
@@ -682,6 +682,25 @@ static int handleCmdVersion( const std::vector<std::string> &args )
...
+  if ( _dupenv_s( &val, &sz, "GEODIFF_LOGGER_LEVEL" ) == 0 && val != nullptr )
+  {
+    ret = atoi( val );
+    free( val );
+  }
+#else
+  char *val = getenv( "GEODIFF_LOGGER_LEVEL" );
+  if ( val )
+    ret = atoi( val );
+#endif
```
Comment (wonder-sk) [inline geodiff/src/geodiff-cli.cpp:700]: why not use getEnvVar / getEnvVarInt?

**Review by wonder-sk — APPROVED** [review]

File: geodiff/src/geodiff-cli.cpp (hunk 2 above)
Comment (PeterPetrik) [reply geodiff/src/geodiff-cli.cpp:700]: I think getEnvVarInt is not in public geodiff.h header, so not sure if I can use it?

File: geodiff/src/changesetvarint.h (hunk 1 above)
Comment (PeterPetrik) [reply geodiff/src/changesetvarint.h:220]: hmm, I forgot why I added it there I think as it is static, in some include these was this function unsused or so

**Pull request merged**